Adding 8 white-space-instrinsic-size tests

TalbotG opened the pull request:
6 min-content sizing tests and 2 max-content sizing tests:

white-space-intrinsic-size-013.html
white-space-intrinsic-size-014.html
white-space-intrinsic-size-015.html
white-space-intrinsic-size-016.html
white-space-intrinsic-size-017.html
white-space-intrinsic-size-018.html
white-space-intrinsic-size-019.html
white-space-intrinsic-size-020.html
reference/ref-filled-green-100px-square.xht

TalbotG committed:
Adding 8 white-space-instrinsic-size tests

frivoal committed:
Reuse rather than duplicate reference file

frivoal commented on css/css-text/white-space/white-space-intrinsic-size-013.html:
```diff
@@ -0,0 +1,77 @@
+  <link rel="stylesheet" type="text/css" href="/fonts/ahem.css">
+
+  <meta content="" name="flags">
+  <meta content="When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line affect the intrinsic min-content size." name="assert">
```
>When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line affect the intrinsic min-content size

That's not correct.
When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line are indeed preserved (and so can be shown if you underline them, but a span around them with a background, etc).
However, according to [4.2](https://drafts.csswg.org/css-text-3/#white-space-phase-2) point 4 bullet 2, they must **hang** (those at the end of wrapped lines hang unconditionally, those before forced breaks hang conditionally, but that difference is not relevant in this test). If you look at the [definition of hanging in 8.2](https://drafts.csswg.org/css-text-3/#hanging), it says:
> The hanging glyph is also not taken into account when computing intrinsic sizes (min-content size and max-content size)

and

> Glyphs that conditionally hang are not taken into account when computing min-content sizes and any sizes derived thereof

So the premise of this test is mistaken, and the green box should be 25px wide (1 letter). Several current implementations are incorrect, so don't trust the results you get in Firefox and Chrome. If I am not mistaken, you'll get correct results for min-content sizing in webkit (not blink) only at the moment.

TalbotG replied:
Thank you for your complete reply. I have redesigned entirely these 2 tests.

frivoal commented on css/css-text/white-space/white-space-intrinsic-size-014.html:
```diff
@@ -0,0 +1,86 @@
+  <link rel="stylesheet" type="text/css" href="/fonts/ahem.css">
+
+  <meta content="" name="flags">
+  <meta content="When 'white-space' is 'pre-wrap', preserved white spaces at the end of the line affect the intrinsic min-content size. Overflow-wrap makes no difference." name="assert">
```
same comment as for -013

frivoal reviewed (changes requested):
Two tests (-013 and -014) need corrections. The rest is good.

TalbotG committed:
Corrected and improved white-space-intrinsic-size-[013-014] tests

TalbotG commented:
I think I now understand the conditionally versus unconditionally 'hang'.
I redesigned completely both tests. I am confident about those 2 tests now.

frivoal committed:
Correct assertion

Since the spaces are before line breaks, they **conditionally** hang.
The assert claimed that they should **unconditionally** hang. However,
in the min-content sizing case, the result is the same, so the test is
correct even if the assertion had was slightlt wrong.

The difference between conditional and unconditional hanging would be
noticeable in a max-content sizing test.

frivoal reviewed (approved):
This is all good now, approved. There was a small mistake in the assertion description, but the behavior of the test is not affected. Please see https://github.com/web-platform-tests/wpt/pull/21967/commits/9884a20b3dbd4b2e33a3581c9388533754115cf0 for details.
